include/seastar/http/exception.hh

```diff
@@ -54,9 +64,14 @@ public:
     virtual const std::string& str() const {
         return _msg;
     }
+
+    const std::string& content_type() const {
+        return _content_type;
+    }
 private:
     std::string _msg;
     http::reply::status_type _status;
+    std::string _content_type;
```

Comment:
Please, initialize it to "json" by default, as this is what really happens when this exception is caught

Reply:
Empty string means something different than json (in the current patch). Here are the two cases:

1) Empty content type: the pre-exisitng behavior of taking the `_msg` (which is _not_ json) and encoding it into the fixed-format json message using ` json_exception(e).to_json()` like `{"code": N, "message": "text"}`.
2) Non-empty content type: we take `_msg` as the body (without transforming it), and the content type as the content type.

So setting this to json would change the default from (1) to (2) and break everything. Specifically we have case 2a) which is when the content type is explicitly "json", the `_msg` should itself be valid json and it returned as the body without further encoding.

Reply:
That's a good suggestion.

I'm going to put this PR on ice for now as I'll be out and not have time to re-work it in the near future, but I may come back to it some day.